lib/handler/RedirectHandler.js

```diff
@@ -91,6 +92,15 @@ class RedirectHandler {
       ? null
       : parseLocation(statusCode, headers)
 
+    if (this.history.length >= this.maxRedirections && !this.redirectionLimitReached) {
+      if (this.request) {
+        this.request.abort()
+      }
+
+      this.redirectionLimitReached = true
```

Comment:
I believe this flag should be handled by the implementation, not necessarily by us (although we can use it as a control flag).

What I mean is, that a use case should look like this:

```js
await request(t, server, undefined, `http://${server}/300`, {
    maxRedirections: 2,
    throwOnMaxRedirect: true
})
``

If `throwOnMaxRedirect` is set to `true`,  whenever we reach this section, we will abort the request with an error (as you are already doing). Otherwise, we continue with the old behaviour (which is to do nothing).

Wdyt?

Reply:
yes that would be much better thank you very much I tried to implement this in my next commit @metcoder95